tools/build_script_generator/cmake/resources/CMakeLists.txt.in

```diff
@@ -37,25 +41,55 @@ add_library(modm OBJECT
 
 %% endif
 
-add_library(modm_warnings INTERFACE)
-add_library(modm_options INTERFACE)
-add_library(modm_arch_options INTERFACE)
+%% if is_modm
+add_library({{ repo }}_warnings INTERFACE)
+add_library({{ repo }}_options INTERFACE)
+add_library({{ repo }}_arch_options INTERFACE)
 
-modm_target_config_create(modm modm_arch_options modm_options modm_warnings)
+modm_target_config_create({{ repo }} {{ repo }}_arch_options {{ repo }}_options {{ repo }}_warnings)
+%% endif
 
 %% if include_paths | length
-target_include_directories(modm SYSTEM PUBLIC
+target_include_directories({{ repo }} SYSTEM
+%% if sources | length
+  PUBLIC
+%% else
+  INTERFACE
+%%endif
 %% for path in include_paths | sort
   {{ path | relocate | modm.posixify  }}
 %% endfor
+%% if sources | length
+  PRIVATE
+%% for path in private_include_paths | sort
+  {{ path | relocate | modm.posixify  }}
+%% endfor
+%%endif
 )
 
 %% endif
 
-target_link_libraries(modm
+
+%% if is_modm
+target_link_libraries({{ repo }}
   PUBLIC
+    {{ repo }}_arch_options
+    project_options
+    {{ repo }}_options
+    {{ repo }}_warnings)
+%% else
```

Comment:
same here?

Reply:
Yeah, the `{{ repo }}` can also just be set to `modm` here.